src/System.Management.Automation/engine/Interop/Windows/CreateFile.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static unsafe partial class Windows
+    {
+        // dwDesiredAccess of CreateFile
+        [Flags]
+        internal enum FileDesiredAccess : uint
+        {
+            GenericZero = 0,
+            GenericRead = 0x80000000,
+            GenericWrite = 0x40000000,
+            GenericExecute = 0x20000000,
+            GenericAll = 0x10000000,
+        }
+
+        // dwFlagsAndAttributes
+        [Flags]
+        internal enum FileAttributes : uint
+        {
+            Readonly = 0x00000001,
+            Hidden = 0x00000002,
+            System = 0x00000004,
+            Archive = 0x00000020,
+            Encrypted = 0x00004000,
+            Write_Through = 0x80000000,
+            Overlapped = 0x40000000,
+            NoBuffering = 0x20000000,
+            RandomAccess = 0x10000000,
+            SequentialScan = 0x08000000,
+            DeleteOnClose = 0x04000000,
+            BackupSemantics = 0x02000000,
+            PosixSemantics = 0x01000000,
+            OpenReparsePoint = 0x00200000,
+            OpenNoRecall = 0x00100000,
+            SessionAware = 0x00800000,
+            Normal = 0x00000080
+        }
+
+        // WARNING: This method does not implicitly handle long paths. Use CreateFile.
+        [LibraryImport("api-ms-win-core-file-l1-1-0.dll", EntryPoint = "CreateFileW", SetLastError = true, StringMarshalling = StringMarshalling.Utf16)]
+        private static unsafe partial SafeFileHandle CreateFilePrivate(
```

Comment:
I believe the intent of the name is that code doesn't call the C function directly

Reply:
@SteveL-MSFT The `private` accessibility modifier should be sufficient.

Reply:
We could have the one extern signature that returns nint and just have the outer methods wrap that in the specific safe handle types.

Reply:
The code was copied from .Net as well tested. Different teams create pinvoke code in .Net so it can have different styles. We have several PRs at the same time so it's also hard for us to follow one style. It is better to make a style correction later.